scripts/get-git-tag-name.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# This script derives a git tag name from the version fields found in a given Go
+# file. It also checks if the derived git tag name is a valid SemVer compliant
+# version string.
+
+# get_git_tag_name reads the version fields from the given file and then
+# constructs and returns a git tag name.
+get_git_tag_name() {
+  local file_path="$1"
+
+  # Check if the file exists
+  if [ ! -f "$file_path" ]; then
+      echo "Error: File not found at $file_path" >&2
+      exit 1
+  fi
+
+  # Read and parse the version fields.
+  local app_major
+  app_major=$(grep -oP 'AppMajor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_minor
+  app_minor=$(grep -oP 'AppMinor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_patch
+  app_patch=$(grep -oP 'AppPatch\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_status
+  app_status=$(grep -oP 'AppStatus\s*=\s*"([a-z]*)"' "$file_path" | cut -d '"' -f 2)
+
+  local app_pre_release
+  app_pre_release=$(grep -oP 'AppPreRelease\s*=\s*"([a-z0-9]*)"' "$file_path" | cut -d '"' -f 2)
+
+  # Parse the GitTagIncludeStatus field.
+  gitTagIncludeStatus=false
+
+  if grep -q 'GitTagIncludeStatus = true' "$file_path"; then
+      gitTagIncludeStatus=true
+  elif grep -q 'GitTagIncludeStatus = false' "$file_path"; then
+      gitTagIncludeStatus=false
+  else
+      echo "Error: GitTagIncludeStatus is not present in the file."
+      exit 1
+  fi
+
+  # Construct the git tag name with conditional inclusion of app_status and
+  # app_pre_release
+  tag_name="v${app_major}.${app_minor}.${app_patch}"
+
+  # Append app_status if gitTagIncludeStatus is true and app_status is not empty
+  if [ "$gitTagIncludeStatus" = true ] && [ -n "$app_status" ]; then
```

Comment:
This doesn't match what we're doing in `version.go`...
For

```
	AppMajor uint = 0
	AppMinor uint = 3
	AppPatch uint = 2
	AppStatus = "alpha"
	AppPreRelease = "rc2"
```
the shell script would produce `v0.3.2-alpha-rc2` while the `version.go` would output `v0.3.2-alpha.rc2`.

Reply:
I've gone with `v0.3.2-alpha.rc2` from both. So the pre-release is prefixed with `.`

Reply:
I think we need another `else if` case for this project where `git_tag_include_status` is `false`. I tried it locally and the tag created was: `v0.3.2.rc1`.

Reply:
OK, we spoke via DM.

My understanding: in the latest commits, `git_tag_include_status` is `false` and `AppPreRelease` is specified then we hope to see git tags of this sort: `v0.3.2-rc1`.

Oli said:
> Yes, anything not part of the actual version number should be separated by a dash. But then the individual parts of the suffix (e.g. alpha and rcX) should be separated by dots to keep the format we have in other projects

Reply:
Looking at our current tags, I think we never use an `rc` suffix without `alpha` or `beta`? And our release tags have neither (though our release page adds `alpha` to the release name & notes).

I understand that if we want to reuse the work in this PR, we don't want to disallow tags with only one suffix set, but for this repo I think we'd want the option to force these suffixes to be linked / paired.

Reply:
We do use the `-alpha.rcX` currently, otherwise the version check in the release script would fail. But IIRC we do **not** want the `alpha` to be in the git tag at all. So now that we use the same logic for the release version check as we do when creating the tag, we can get rid of it.

Reply:
Double checked the golang rules, I agree that we could satisfy those rules without the `alpha` or `beta` keywords.